app/src/main/kotlin/com/simplemobiletools/calculator/activities/CryptoActivity.kt

```diff
@@ -0,0 +1,149 @@
+    private var storedUseEnglish = false
+
+    private var cryptoFROM: String = ""
+    private var cryptoTO: String = ""
+
+    lateinit var calc: CryptoCalculatorImpl
+
+    @SuppressLint("MissingSuperCall")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_crypto)
+
+        calc = CryptoCalculatorImpl(this, applicationContext)
+        updateViewColors(crypto_holder, config.textColor)
+
+        getButtonIds().forEach {
+            it.setOnClickListener {
+                calc.numpadClicked(it.id); checkHaptic(it)
+            }
+        }
+
+        var numberPickerFROM: NumberPicker = numberPicker1
+        var numberPickerTO: NumberPicker = numberPicker2
+
+        // Values of each of the Number Pickers
+        val valuesFROM = arrayOf("Bitcoin", "Ethereum", "Ripple", "Litecoin", "USD")
+        val valuesTO = arrayOf("Ethereum", "Ripple", "Litecoin", "USD", "Bitcoin")
+
+        // Setting default value of each Number Picker
+        numberPickerFROM.minValue = 0
+        numberPickerTO.minValue = 0
+
+        // Setting size of each Number Picker
+        numberPickerFROM.maxValue = valuesFROM.size - 1
+        numberPickerTO.maxValue = valuesTO.size - 1
+
+        // Setting Number Pickers to the values
+        numberPickerFROM.displayedValues = valuesFROM
+        numberPickerTO.displayedValues = valuesTO
+
+        // Sets whether the selector wheel wraps when reaching the min/max value.
+        numberPickerFROM.wrapSelectorWheel = true
+        numberPickerTO.wrapSelectorWheel = true
+
+        val cryptoSymbolHashMap = HashMap<String, String>()
+        cryptoSymbolHashMap.put("Ethereum", "ETH")
+        cryptoSymbolHashMap.put("Bitcoin", "BTC")
+        cryptoSymbolHashMap.put("Ripple", "XRP")
+        cryptoSymbolHashMap.put("Litecoin", "LTC")
+        cryptoSymbolHashMap.put("USD", "USD")
+
+        btn_convert_cryto.setOnClickListener { _ ->
+            cryptoFROM = cryptoSymbolHashMap.get(valuesFROM[numberPickerFROM.value]).toString()
+            cryptoTO = cryptoSymbolHashMap.get(valuesTO[numberPickerTO.value]).toString()
+
+            if (cryptoTO == cryptoFROM) {
+                calc.overwriteNumber(Formatter.stringToDouble(getResult()))
+            } else {
+                performConversion(cryptoFROM, cryptoTO)
+            }
+        }
+        btn_delete.setOnClickListener { calc.handleDelete(); checkHaptic(it) }
+        btn_delete.setOnLongClickListener { calc.handleClear(); true }
+        result.setOnLongClickListener { copyToClipboard(result.value); true }
+
+        AutofitHelper.create(result)
+    }
+
+    private fun performConversion(cryptoFROM: String, cryptoTO: String) {
+        if (isOnline()) {
+            calc.calculateCrypto(cryptoFROM, cryptoTO, this)
+        } else if (!isOnline()) {
+            displayToast("It seems there has been a connection problem, contact you ISP for more details !")
```

Comment:
Maybe we should consider displaying an alertDialog to the user. This toast is too easy to dismiss. Perhaps something similar using the code for my error handling feature.

Reply:
[PR for reference](https://github.com/jusleg/simple-calculator-FAMINGO/pull/155)